File: python/kode program/elektronik.py
from produk import Produk# Mengimpor class Produk dari file produk.py

class Elektronik(Produk):# Deklarasi class Elektronik, turunan dari Produk
    def __init__(self, id, nama, harga, kategori, merk, garansi, stok, warna):
        super().__init__(id, nama, harga, kategori)  # Memanggil constructor class Produk
        self.__merk = merk# Merk produk elektronik (atribut private)
        self.__garansi = garansi# Lama garansi (atribut private)
        self.__stok = stok# Jumlah stok produk (atribut private)
        self.__warna = warna# Warna produk (atribut private)

    # Getter & Setter
    def get_merk(self): return self.__merk
    def set_merk(self, value): self.__merk = value

    def get_garansi(self): return self.__garansi
    def set_garansi(self, value): self.__garansi = value

    def get_stok(self): return self.__stok
    def set_stok(self, value): self.__stok = value

    def get_warna(self): return self.__warna
    def set_warna(self, value): self.__warna = value

File: java/kode program/Elektronik.java
public class Elektronik extends Produk {  // Class Elektronik turunan dari Produk
    protected String merk;// Merk produk
    protected int garansi;// Lama garansi 
    protected int stok;// Jumlah stok produk
    protected String warna;// Warna produk

    // Constructor dengan parameter
    public Elektronik(int id,String nama,double harga,String kategori,
                      String merk,int garansi,int stok,String warna) {
        super(id,nama,harga,kategori); // Panggil constructor Produk
        this.merk=merk; this.garansi=garansi; this.stok=stok; this.warna=warna;
    }

    // Getter & Setter
    public String getMerk() { return merk; }
    public void setMerk(String merk) { this.merk=merk; }

    public int getGaransi() { return garansi; }
    public void setGaransi(int garansi) { this.garansi=garansi; }

    public int getStok() { return stok; }
    public void setStok(int stok) { this.stok=stok; }

    public String getWarna() { return warna; }
    public void setWarna(String warna) { this.warna=warna; }
}
